### PR TITLE
fs/vfs: Add vopen

### PR DESCRIPTION
**Pull request by junmin-kim**
vopen() is an internal TizenRT interface

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [6060ce87cf1cd38129a93dbbb945252aa131001d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6060ce87cf1cd38129a93dbbb945252aa131001d] - Code Rule Check (C++) OK.

**Comment by junmin-kim** [comment]
This PR is under testing

**Comment by seinfra** [comment]
Target : [3571f6d03a97f184851c80c7bbb0783a0a273c70] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3571f6d03a97f184851c80c7bbb0783a0a273c70] - Code Rule Check (C++) OK.

File: os/fs/vfs/fs_open.c [hunk 1]
```diff
@@ -97,14 +93,22 @@ int inode_checkflags(FAR struct inode *inode, int oflags)
...
 /****************************************************************************
- * Name: open
+ * Name: vopen
  *
  * Description:
- *   Standard 'open' interface
+ *   vopen() is identical to 'open' except that it accepts a va_list
+ *   as an argument versus taking a variable length list of arguments.
+ *
+ *   vopen() is an internal TizenRT interface and should not be called from
```
Comment (chanijjani) [inline os/fs/vfs/fs_open.c]: If this function is internal, how about locating this above line 74 after recovering  /** Private Functions ***/?

File: os/fs/vfs/fs_open.c [hunk 2]
```diff
@@ -164,11 +168,10 @@ int open(const char *path, int oflags, ...)
 		/* Get the file descriptor of the opened character driver proxy */
 		fd = block_proxy(path, oflags);
 		if (fd < 0) {
-			ret = -fd;
+			ret = fd;
```
Comment (chanijjani) [inline os/fs/vfs/fs_open.c]: To return a negative fd is meaningful for debugging..? It seems that to return a specific errno value (negated) is better.

File: os/include/tinyara/fs/fs.h [hunk 3]
```diff
@@ -741,6 +741,23 @@ int lib_flushall(FAR struct streamlist *list);
...
+ *
+ *   vopen() is an internal TizenRT interface and should not be called from
+ *   applications.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned on
+ *   any failure.
+ *
+ ****************************************************************************/
+int vopen(FAR const char *path, int oflags, va_list ap);
```
Comment (chanijjani) [inline os/include/tinyara/fs/fs.h]: I just wonder why this function name starts with "v".. for video?

Comment (junmin-kim) [reply os/include/tinyara/fs/fs.h]: If a function starts with v, it passes a variant argument.

File: os/fs/vfs/fs_open.c (hunk 2 above)
Comment (junmin-kim) [reply os/fs/vfs/fs_open.c]: If `block_proxy` returns a negative value, it may be ENOMEM.

File: os/fs/vfs/fs_open.c (hunk 1 above)
Comment (junmin-kim) [reply os/fs/vfs/fs_open.c]: Currently, we are porting functions we need. To avoiding conflict, we follow the same sequence as Nuttx.

**Comment by seinfra** [comment]
Target : [438abcdb5f6c53fa79aeca126ac13a0fbcb5171d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [438abcdb5f6c53fa79aeca126ac13a0fbcb5171d] - Code Rule Check OK.

File: os/fs/vfs/fs_open.c [hunk 4]
```diff
@@ -115,18 +119,18 @@ int open(const char *path, int oflags, ...)
...
 #endif
 
-	/* open() is a cancellation point */
-	(void)enter_cancellation_point();
-
 	/* If the file is opened for creation, then get the mode bits */
 
 	if ((oflags & (O_WRONLY | O_CREAT)) != 0) {
-		va_list ap;
 		va_start(ap, oflags);
```
Comment (zhouxinhe) [inline os/fs/vfs/fs_open.c]: ‘va_start’ was used in function vopen with fixed args.
I think we should use va_arg directly.
@junmin-kim

**Pull request merged**